src/app/stores/migrator_baseline.ts

```diff
@@ -0,0 +1,278 @@
+    glyph.constraints.forEach(constraint => {
+      if (constraint.type === "snap") {
+        const id = constraint.attributes.targetElement as string;
+        const guide = guides[id];
+        if (guide && constraint.attributes.targetAttribute === "value2") {
+          // make a new guide
+          const newGuide = createGuide(
+            guide.properties.axis as GuideAxis,
+            guide,
+            +guide.properties.value + +guide.properties.gap
+          );
+          // add new guide
+          glyph.marks.push(newGuide.element);
+          // add state instances
+          related.forEach(ref => {
+            const s = ref.state as PlotSegmentState;
+            if (s && s.glyphs) {
+              s.glyphs.forEach(glyphState => {
+                glyphState.marks.push(newGuide.state);
+              });
+            }
+          });
+          if (nested) {
+            // nested charts store in mappings
+            const valueMapping: ValueMapping = {
+              type: "value",
+              value: newGuide.state.attributes.value
+            };
+            newGuide.element.mappings.value = valueMapping;
+          }
+          // point to new guide
+          constraint.attributes.targetElement = newGuide.element._id;
+          constraint.attributes.targetAttribute =
+            GuideAttributeNames.computedBaselineValue;
+        }
+      }
+    });
+
+    // if (guide.mappings) {
+    // TODO guides should not be mapped!
+    // }
+
+    for (const _id in guides) {
+      const guide = guides[_id];
+      // add new properties to guide
+      addNewGuideProperties(guide);
+      // delete old properties
+      removeOldGuideProperties(guide);
+      // modify all state instances
+      related.forEach(ref => {
+        const s = ref.state as PlotSegmentState;
+        if (s && s.glyphs) {
+          s.glyphs.forEach(glyphState => {
+            glyphState.marks.forEach(markState => {
+              // add new properties to guide
+              addNewGuideProperties(null, markState);
+              // delete old properties
+              removeOldGuideProperties(null, markState);
+            });
+          });
+        }
+      });
+    }
+  });
+}
+
+function find(
+  elements: Element[],
+  states: ElementState[],
+  predicate: (element: Element) => boolean
+) {
+  const refs: ElementRef[] = [];
+  elements.forEach((element, index) => {
+    if (predicate(element)) {
+      const state = states && states[index];
+      refs.push({ element, index, state });
+    }
+  });
+  return refs;
+}
+
+function changeConstraintTarget(
+  element: Element,
+  constraint: Specification.Constraint,
+  guideValue: number,
+  elementCollection: Element[],
+  stateCollection: ElementState[]
+) {
+  const gap = +element.properties.gap;
+  if (constraint.attributes.targetAttribute === "value2" && gap) {
+    // create a 2nd guide to insert, based on gap property of first
+    const axis = element.properties.axis as GuideAxis;
```

Comment:
Is this always true? i.e. Will the `axis` property here always be a **GuideAxis**?

Reply:
If not, it will default to "center".